include/wil/network.h

```diff
@@ -0,0 +1,1328 @@
+//*********************************************************
+//
+//    Copyright (c) Microsoft. All rights reserved.
+//    This code is licensed under the MIT License.
+//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+//    ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+//    TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+//    PARTICULAR PURPOSE AND NONINFRINGEMENT.
+//
+//*********************************************************
+//! @file
+//! Helpers for using BSD sockets and Winsock functions and structures.
+//! Does not require the use of the STL or C++ exceptions (see _nothrow functions)
+#ifndef __WIL_NETWORK_INCLUDED
+#define __WIL_NETWORK_INCLUDED
+
+#ifdef _KERNEL_MODE
+#error This header is not supported in kernel-mode.
+#endif
+
+//! define WIN32_LEAN_AND_MEAN at the project level to avoid windows.h including older winsock 1.1 headers from winsock.h

```

Comment:
Not documenting anything so don't use doc comments (i.e. use `//` instead of `//!`). This comment applies throughout the PR